library/insights_registration.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2018, Terry Jones <terry.jones@example.org>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: insights_registration
+
+short_description: This module registers the insights client
+
+description:
+    - This module will check the current registration status, unregister if needed,
+    and then register the insights client (and update the display_name if needed)
+
+options:
+    insights_name:
+        description:
+            - For now, this is just 'insights-client', but it could change in the future
+            so having it as a variable is just preparing for that
+        required: true
+    display_name:
+        description:
+            - When registering with insights an optional display_name can be configured
+        required: false
+
+author:
+    - Jason Stephens (@Jason-RH)
+'''
+
+EXAMPLES = '''
+# Register a fresh install
+- name: Register the insights client on a fresh install
+  insights_registration:
```

Comment:
I was about to suggest renaming to `insights_register`, but I realize now you are consistent with the style used in another similar moduel (e.g. `redhat_subscription`).  Thanks!

Reply:
I actually did rename it to insights_register, just haven't committed. Do you recommend keeping it that way?

Reply:
Your call, whichever you feel more strongly about.  Thanks!